### PR TITLE
Fix undefined method `owners' for NullPreloader:Class

### LINKED ISSUE #24413: NoMethodError: undefined method `owners' for ActiveRecord::Associations::Preloader::NullPreloader:Class
### Steps to reproduce

https://gist.github.com/Ladas/f066670b2a6ebabbe40c7d49ecd37dd3

### Expected behavior

Tests in the gist ^ should pass

### Actual behavior

Tests in the gist are not passing :-)

### PR DESCRIPTION
**Pull request by Ladas**
Fixes https://github.com/rails/rails/issues/24413
### Summary

Fixing undefined method `owners' for ActiveRecord::Associations::Preloader::NullPreloader:Class

This a building of a complex query with include, that is getting this result. The root of the problem is probably more :through relations, when one link of the chain is empty
### Other Information

This fix is probably not right, might be that ActiveRecord::Associations::Preloader::NullPreloader:Class should just have :owners method defined?
### Application specific details

This happen when I combine my 2 PRs:
https://github.com/ManageIQ/manageiq/pull/7033
https://github.com/ManageIQ/manageiq/pull/7237

This brings extra :through layer https://github.com/ManageIQ/manageiq/pull/7237, then I start to see this error

StackTrace of the ManageIQ app where this occurs:

```
[----] I, [2016-03-14T19:21:26.529830 #16851:41dc71c]  INFO -- : Started GET "/cloud_subnet/show_list" for ::1 at 2016-03-14 19:21:26 +0100
[----] I, [2016-03-14T19:21:27.254761 #16851:41dc71c]  INFO -- : Processing by CloudSubnetController#show_list as HTML
[----] F, [2016-03-14T19:21:31.411707 #16851:41dc71c] FATAL -- : Error caught: [NoMethodError] undefined method `owners' for ActiveRecord::Associations::Preloader::NullPreloader:Class
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/associations/preloader/through_association.rb:36:in `block in associated_records_by_owner'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/associations/preloader/through_association.rb:35:in `each'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/associations/preloader/through_association.rb:35:in `each_with_object'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/associations/preloader/through_association.rb:35:in `associated_records_by_owner'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/associations/preloader/has_many_through.rb:8:in `associated_records_by_owner'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/associations/preloader/collection_association.rb:8:in `preload'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/associations/preloader/association.rb:19:in `run'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/associations/preloader.rb:151:in `block (2 levels) in preloaders_for_one'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/associations/preloader.rb:149:in `each'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/associations/preloader.rb:149:in `map'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/associations/preloader.rb:149:in `block in preloaders_for_one'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/associations/preloader.rb:148:in `each'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/associations/preloader.rb:148:in `flat_map'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/associations/preloader.rb:148:in `preloaders_for_one'
/home/Ladas/Projects/cfme/manageiq/lib/extensions/ar_virtual.rb:263:in `preloaders_for_one'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/associations/preloader.rb:115:in `preloaders_on'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/associations/preloader.rb:102:in `block in preload'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/associations/preloader.rb:101:in `each'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/associations/preloader.rb:101:in `flat_map'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/associations/preloader.rb:101:in `preload'
/home/Ladas/Projects/cfme/manageiq/lib/extensions/ar_virtual.rb:258:in `block in preloaders_for_one'
/home/Ladas/Projects/cfme/manageiq/lib/extensions/ar_virtual.rb:256:in `each'
/home/Ladas/Projects/cfme/manageiq/lib/extensions/ar_virtual.rb:256:in `flat_map'
/home/Ladas/Projects/cfme/manageiq/lib/extensions/ar_virtual.rb:256:in `preloaders_for_one'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/associations/preloader.rb:125:in `block in preloaders_for_hash'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/associations/preloader.rb:124:in `each'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/associations/preloader.rb:124:in `flat_map'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/associations/preloader.rb:124:in `preloaders_for_hash'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/associations/preloader.rb:113:in `preloaders_on'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/associations/preloader.rb:102:in `block in preload'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/associations/preloader.rb:101:in `each'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/associations/preloader.rb:101:in `flat_map'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/associations/preloader.rb:101:in `preload'
/home/Ladas/Projects/cfme/manageiq/lib/miq_preloader.rb:4:in `preload'
/home/Ladas/Projects/cfme/manageiq/lib/extensions/ar_virtual.rb:289:in `find_with_associations'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/relation.rb:699:in `exec_queries'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/relation.rb:580:in `load'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/relation.rb:260:in `records'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/relation/delegation.rb:39:in `length'
/home/Ladas/Projects/cfme/manageiq/app/models/rbac.rb:292:in `find_targets_without_rbac'
/home/Ladas/Projects/cfme/manageiq/app/models/rbac.rb:287:in `find_targets_with_rbac'
/home/Ladas/Projects/cfme/manageiq/app/models/rbac.rb:507:in `search'
/home/Ladas/Projects/cfme/manageiq/app/models/miq_report/search.rb:109:in `paged_view_search'
/home/Ladas/Projects/cfme/manageiq/app/controllers/application_controller.rb:1771:in `get_view'
/home/Ladas/Projects/cfme/manageiq/app/controllers/application_controller/ci_processing.rb:1026:in `process_show_list'
/home/Ladas/Projects/cfme/manageiq/app/controllers/mixins/generic_list_mixin.rb:8:in `show_list'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionpack/lib/action_controller/metal/basic_implicit_render.rb:4:in `send_action'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionpack/lib/abstract_controller/base.rb:181:in `process_action'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionpack/lib/action_controller/metal/rendering.rb:30:in `process_action'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionpack/lib/abstract_controller/callbacks.rb:20:in `block in process_action'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activesupport/lib/active_support/callbacks.rb:126:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activesupport/lib/active_support/callbacks.rb:126:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activesupport/lib/active_support/callbacks.rb:506:in `block (2 levels) in compile'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activesupport/lib/active_support/callbacks.rb:455:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activesupport/lib/active_support/callbacks.rb:455:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activesupport/lib/active_support/callbacks.rb:101:in `__run_callbacks__'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activesupport/lib/active_support/callbacks.rb:750:in `_run_process_action_callbacks'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activesupport/lib/active_support/callbacks.rb:90:in `run_callbacks'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionpack/lib/abstract_controller/callbacks.rb:19:in `process_action'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionpack/lib/action_controller/metal/rescue.rb:31:in `process_action'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionpack/lib/action_controller/metal/instrumentation.rb:32:in `block in process_action'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activesupport/lib/active_support/notifications.rb:164:in `block in instrument'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activesupport/lib/active_support/notifications/instrumenter.rb:21:in `instrument'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activesupport/lib/active_support/notifications.rb:164:in `instrument'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionpack/lib/action_controller/metal/instrumentation.rb:30:in `process_action'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionpack/lib/action_controller/metal/params_wrapper.rb:248:in `process_action'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activerecord/lib/active_record/railties/controller_runtime.rb:18:in `process_action'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionpack/lib/abstract_controller/base.rb:126:in `process'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionview/lib/action_view/rendering.rb:30:in `process'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionpack/lib/action_controller/metal.rb:190:in `dispatch'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionpack/lib/action_controller/metal.rb:262:in `dispatch'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionpack/lib/action_dispatch/routing/route_set.rb:50:in `dispatch'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionpack/lib/action_dispatch/routing/route_set.rb:32:in `serve'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionpack/lib/action_dispatch/journey/router.rb:39:in `block in serve'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionpack/lib/action_dispatch/journey/router.rb:26:in `each'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionpack/lib/action_dispatch/journey/router.rb:26:in `serve'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionpack/lib/action_dispatch/routing/route_set.rb:739:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/gems/omniauth-1.3.1/lib/omniauth/strategy.rb:186:in `call!'
/home/Ladas/.rvm/gems/ruby-2.2.3/gems/omniauth-1.3.1/lib/omniauth/strategy.rb:164:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/gems/omniauth-1.3.1/lib/omniauth/builder.rb:63:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/gems/secure_headers-3.0.3/lib/secure_headers/middleware.rb:10:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionview/lib/action_view/digestor.rb:12:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/gems/rack-2.0.0.alpha/lib/rack/etag.rb:25:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/gems/rack-2.0.0.alpha/lib/rack/conditional_get.rb:25:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/gems/rack-2.0.0.alpha/lib/rack/head.rb:12:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/gems/rack-2.0.0.alpha/lib/rack/session/abstract/id.rb:220:in `context'
/home/Ladas/.rvm/gems/ruby-2.2.3/gems/rack-2.0.0.alpha/lib/rack/session/abstract/id.rb:214:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionpack/lib/action_dispatch/middleware/cookies.rb:613:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionpack/lib/action_dispatch/middleware/callbacks.rb:38:in `block in call'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activesupport/lib/active_support/callbacks.rb:97:in `__run_callbacks__'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activesupport/lib/active_support/callbacks.rb:750:in `_run_call_callbacks'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activesupport/lib/active_support/callbacks.rb:90:in `run_callbacks'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionpack/lib/action_dispatch/middleware/callbacks.rb:36:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionpack/lib/action_dispatch/middleware/executor.rb:12:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionpack/lib/action_dispatch/middleware/remote_ip.rb:79:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionpack/lib/action_dispatch/middleware/debug_exceptions.rb:49:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionpack/lib/action_dispatch/middleware/show_exceptions.rb:31:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/railties/lib/rails/rack/logger.rb:36:in `call_app'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/railties/lib/rails/rack/logger.rb:26:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/gems/request_store-1.3.0/lib/request_store/middleware.rb:9:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionpack/lib/action_dispatch/middleware/request_id.rb:24:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/gems/rack-2.0.0.alpha/lib/rack/method_override.rb:22:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/gems/rack-2.0.0.alpha/lib/rack/runtime.rb:22:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/activesupport/lib/active_support/cache/strategy/local_cache_middleware.rb:28:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionpack/lib/action_dispatch/middleware/executor.rb:12:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/actionpack/lib/action_dispatch/middleware/static.rb:136:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/gems/rack-2.0.0.alpha/lib/rack/sendfile.rb:111:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/bundler/gems/rails-feef64cee449/railties/lib/rails/engine.rb:522:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/gems/puma-2.16.0/lib/puma/server.rb:557:in `handle_request'
/home/Ladas/.rvm/gems/ruby-2.2.3/gems/puma-2.16.0/lib/puma/server.rb:404:in `process_client'
/home/Ladas/.rvm/gems/ruby-2.2.3/gems/puma-2.16.0/lib/puma/server.rb:270:in `block in run'
/home/Ladas/.rvm/gems/ruby-2.2.3/gems/puma-2.16.0/lib/puma/thread_pool.rb:106:in `call'
/home/Ladas/.rvm/gems/ruby-2.2.3/gems/puma-2.16.0/lib/puma/thread_pool.rb:106:in `block in spawn_thread'
[----] I, [2016-03-14T19:21:31.420386 #16851:41dc71c]  INFO -- :   Rendering layouts/exception.html.haml within layouts/application
[----] I, [2016-03-14T19:21:31.430582 #16851:41dc71c]  INFO -- :   Rendered layouts/_exception_contents.html.haml (5.0ms)
[----] I, [2016-03-14T19:21:31.430759 #16851:41dc71c]  INFO -- :   Rendered layouts/exception.html.haml within layouts/application (10.2ms)
[----] I, [2016-03-14T19:21:31.441036 #16851:41dc71c]  INFO -- :   Rendered layouts/_doctype.html.haml (1.7ms)
[----] I, [2016-03-14T19:21:34.194908 #16851:41dc71c]  INFO -- :   Rendered stylesheets/_template50.html.haml (2.3ms)
[----] I, [2016-03-14T19:21:35.033150 #16851:41dc71c]  INFO -- :   Rendered layouts/_i18n_js.html.haml (3.8ms)
[----] I, [2016-03-14T19:21:35.066909 #16851:41dc71c]  INFO -- :   Rendered layouts/_user_options.html.haml (13.9ms)
[----] I, [2016-03-14T19:21:35.092621 #16851:41dc71c]  INFO -- :   Rendered layouts/_page_header_navbar.html.haml (25.0ms)
[----] I, [2016-03-14T19:21:35.095483 #16851:41dc71c]  INFO -- :   Rendered layouts/_spinner.html.haml (2.2ms)
[----] I, [2016-03-14T19:21:35.097470 #16851:41dc71c]  INFO -- :   Rendered layouts/_lightbox_panel.html.haml (1.4ms)
[----] I, [2016-03-14T19:21:35.097624 #16851:41dc71c]  INFO -- :   Rendered layouts/_header.html.haml (63.7ms)
[----] I, [2016-03-14T19:21:35.115346 #16851:41dc71c]  INFO -- :   Rendered layouts/_breadcrumbs.html.haml (2.2ms)
[----] I, [2016-03-14T19:21:35.115658 #16851:41dc71c]  INFO -- :   Rendered layouts/_content.html.haml (17.4ms)
[----] I, [2016-03-14T19:21:35.123156 #16851:41dc71c]  INFO -- :   Rendered layouts/_adv_search.html.haml (4.7ms)
[----] I, [2016-03-14T19:21:35.123374 #16851:41dc71c]  INFO -- :   Rendered layouts/_footer.html.haml (7.1ms)
[----] I, [2016-03-14T19:21:35.123853 #16851:41dc71c]  INFO -- : Completed 500 Internal Server Error in 7869ms (Views: 3711.1ms | ActiveRecord: 0.0ms)
```

### CODE REVIEW

**Comment by rafaelfranca** [comment]
The best option would be define `owners` in that class but we need to understand why you are getting a `NullPreloader` in your query. Were you able to reproduce this error in the Rails test suite?

**Comment by Ladas** [comment]
@rafaelfranca not yet, I guess it will take me some time, it's a chain of a lot of :through relations with also polymorphic relations put as an include to even more :through relations. Somewhere in that the NullPreloader appears. :-)

**Comment by Ladas** [comment]
@rafaelfranca so adding the 'owners' class method in the NullPreloader fixes the issue

**Comment by rafaelfranca** [comment]
Awesome! Now we need to write a test for this.

**Comment by Ladas** [comment]
@rafaelfranca right, I'll try to do that next week

**Comment by Ladas** [comment]
@rafaelfranca here is the reproducer for this https://github.com/rails/rails/issues/24413, is it also a test? OR I should write this as a test somewhere?

**Comment by rafaelfranca** [comment]
You can try to put this test in the Active Record suite. I can give it a try too if you don't want.

**Comment by Ladas** [comment]
@rafaelfranca let me try today, if I will fail horribly, I will be glad for a help :-)

**Comment by Ladas** [comment]
@rafaelfranca ok, tests added. I had to add one more many to many table, to reproduce. But now the owner of the pet knows who is the looter of the pet's treasure. :-D

File: activerecord/test/cases/associations/has_many_through_associations_test.rb [hunk 1]
```diff
@@ -1082,6 +1084,18 @@ def test_preloading_empty_through_association_via_joins
...
+    owner = Owner.create!(:name => "Rainbow Unicat")
+    pet = Pet.create!(:owner => owner)
+    person = Person.create!(:first_name => "Gaga")
+    treasure = Treasure.create!(:looter => person)
+    non_looted_treasure = Treasure.create!()
+    PetTreasure.create!(:pet => pet, :treasure => treasure, :rainbow_color => "Ultra violet indigo")
+    PetTreasure.create!(:pet => pet, :treasure => non_looted_treasure, :rainbow_color => "Ultra violet indigo")
+
+    assert_equal [person], Owner.where(:name => "Rainbow Unicat").includes(:pets => :persons).first.persons.to_a
+  end
```
Comment (vipulnsward) [inline activerecord/test/cases/associations/has_many_through_associations_test.rb]: @Ladas can you use 1.9 hash syntax everywhere.

Comment (Ladas) [inline activerecord/test/cases/associations/has_many_through_associations_test.rb]: @vipulnsward AFAIK I am doing that. :-) Or which one do you mean?

Comment (vipulnsward) [inline activerecord/test/cases/associations/has_many_through_associations_test.rb]: Ex:
```ruby 
Owner.where(name: "Rainbow Unicat").includes(pets: :persons)
PetTreasure.create!(pet: pet, treasure: treasure, rainbow_color: "Ultra violet indigo")
```

Comment (Ladas) [inline activerecord/test/cases/associations/has_many_through_associations_test.rb]: @vipulnsward ah I thought 1.9 is a =>, seems like rest of the file has =>, so I thought that being consistent is the best. :-)  Ok, let me change it

Comment (Ladas) [inline activerecord/test/cases/associations/has_many_through_associations_test.rb]: fixed

Comment (Ladas) [inline activerecord/test/cases/associations/has_many_through_associations_test.rb]: hm, this pets: :persons, looks funky :-)

**Pull request merged**